controllers/places.js

```diff
@@ -0,0 +1,111 @@
+var _ = require('underscore'),
+    db = require('../db');
+
+var getPlace = function(id, callback) {
+  db.medic.get(id, function(err, doc) {
+    if (err) {
+      if (err.statusCode === 404) {
+        err.message  = 'Failed to find place.';
+      }
+      return callback(err);
+    }
+    callback(null, doc);
+  });
+};
+
+var isAPlace = function(place) {
+  return [
+    'district_hospital',
+    'health_center',
+    'clinic'
+  ].indexOf(place.type) !== -1;
+};
+
+/*
+ * Validate the basic data structure for a place.  Not checking against the
+ * database if parent exists because the entire child-parent structure might be
+ * created in one request. Just checking `type` field values and some required
+ * fields.
+ */
+var validatePlace = function(place, callback) {
+  if (!_.isObject(place)) {
+    return callback('Place must be an object.');
+  }
+  if (!isAPlace(place)) {
+    return callback('Wrong type, this is not a place.');
+  }
+  if (_.isUndefined(place.name)) {
+    return callback('Name property missing on place.');
+  }
+  if (!_.isString(place.name)) {
+    return callback('Property "name" must be a string.');
+  }
+  if (place.parent) {
+    // validate parents also
+    return validatePlace(place.parent, callback);
+  }
```

Comment:
Maybe validate that place.parent exists unless this is a district_hospital? I don't think it'd be "correct" to have a clinic without a health_centre for example.